File: node-ui/classes/Database/TwitterHashtagsAccounts.js
const { DataTypes } = require('sequelize');
const sequelize = require('./base');
const TwitterAccounts = require('./TwitterAccounts');

const TwitterHashtagsAccounts = sequelize.define('twitter_hashtags_accounts', {
    id: {
        type: DataTypes.INTEGER(11),
        allowNull: false,
        primaryKey: true,
        autoIncrement: true
    },
    hashtag: {
        type: DataTypes.STRING(128),
        allowNull: false
    },
    date: {
        type: DataTypes.DATEONLY,
        allowNull: false
    },
    handle: {
        type: DataTypes.STRING(255),
        allowNull: false,
        references: {
            model: TwitterAccounts,
            key: 'handle',
        },
        onDelete: 'CASCADE'
    }
});

TwitterHashtagsAccounts.belongsTo(TwitterAccounts, {foreignKey: 'handle', targetKey: 'handle', onDelete: 'CASCADE'});

module.exports = TwitterHashtagsAccounts

File: py-crawler/classes/databases/twitter_hashtags_accounts.py
from sqlalchemy import Column
from sqlalchemy import String, Date, Integer, ForeignKey

from .base import Base

# twitter_hashtags_accounts table.
# Author: Antonio Scardace

class TwitterHashtagsAccounts(Base):

    __tablename__ = 'twitter_hashtags_accounts'

    id = Column('id', Integer, nullable=False, primary_key=True)
    handle = Column('handle', String(255), ForeignKey('twitter_accounts.handle', ondelete='CASCADE'), nullable=False)
    hashtag = Column('hashtag', String(128), nullable=False)
    date = Column('date', Date, nullable=False)

    def __init__(self, id: str):
        self.id = id